Finished documenting the shards in linalg.cpp

ShahFragcolor opened the pull request:
- Added/refined all the shards in linalg.cpp
- Removed the None type for the Flip Y parameter for Project and Unproject shards, gave it a default true instead if not set/
- Made Cross into a struct so that it can have specific help text.


<!--
Before submitting your PR, please review the following checklist:

- [ ] **DO NOT** submit a PR without having discussed the change first in a related issue. If none exist, create one first.
- [ ] **CONSIDER** adding a unit test if your PR resolves an issue.
- [ ] **DO** keep pull requests small so they can be easily reviewed.
- [ ] **DO** make sure all unit tests pass.
- [ ] **DO** make sure not to introduce any new compiler warnings.
- [ ] **AVOID** breaking the continuous integration build.
- [ ] **AVOID** making significant changes to the overall architecture.
-->

ShahFragcolor committed:
Finished documenting the shards in linalg.cpp

- Added/refined all the shards in linalg.cpp
- Removed the None type for the Flip Y parameter for Project and Unproject shards, gave it a default true instead if not set/
- Made Cross into a struct so that it can have specific help text.

ShahFragcolor committed:
Finished doucmentation for Audio shards in audio.cpp

-  Added help text for all audio shards
- Audio.Position, Audio.Direction, Audio.Cones and Audio.Velocity changed the output type from Float to Float3, as they are suppose to output their input unchanged
- Made it such that Audio.Position, Audio.Direction, Audio.Cones, Audio.Velocity, Audio.Pitch, Audio.Volume, Audio.Pan can now accept a context variable for its Sound: parameter
- Added tests for Audio.Position, Audio.Direction, Audio.Cones, Audio.Velocity, Audio.Pitch, Audio.Volume, Audio.Pan in test script

sinkingsugar commented:
You some compilation errors, redeclaring methods or so

sinkingsugar reviewed (approved):
Need to fix tests, looks good, every statement verified thru ai?

ShahFragcolor committed:
Fixed linalg test

- Math.Cross erroneously did not handle float3 sequences, and hence the test was throwing an error. Fixed it to accept float3 sequences as well

ShahFragcolor committed:
More lin alg fixes
